matcher.go

```diff
@@ -9,47 +9,38 @@ type StringMatch struct {
 	test        string
 	invertMatch bool
 	prefixMatch bool
-	isRegex     bool
 	regex       *regexp.Regexp
 }
 
 type StringMatchList []StringMatch
 
 func NewStringMatch(s string) StringMatch {
-
-	invert := false
-	if strings.HasPrefix(s, "!") {
-		invert = true
+	prefix := false
+	invert := strings.HasPrefix(s, "!")
+	if invert {
 		s = s[1:]
 	}
 
-	regex := false
 	var compiledRegex *regexp.Regexp = nil
 	if strings.HasPrefix(s, "regex:") {
-		regex = true
 		s = s[6:]
 		compiledRegex = regexp.MustCompile(s)
-	}
-	prefix := false
-	if !regex {
-		if strings.HasSuffix(s, "*") {
-			prefix = true
-			s = s[0 : len(s)-1]
-		}
+	} else if strings.HasSuffix(s, "*") {
+		prefix = true
+		s = s[0 : len(s)-1]
```

Comment:
FYI `0` can be omitted. Also, is this formatted by gofmt?

Reply:
It is, which honestly surprised me.